components/renku_data_services/project/core.py

```diff
@@ -182,6 +186,29 @@ def validate_session_secrets_patch(
     return result
 
 
+def _validate_repositories(repositories: list[str] | None) -> list[str] | None:
+    """Validate a list of git repositories."""
+    if repositories is None:
+        return None
+    seen: set[str] = set()
+    without_duplicates: list[str] = []
+    for repo in repositories:
+        repo = _validate_repository(repo)
+        if repo not in seen:
+            without_duplicates.append(repo)
+            seen.add(repo)
+    return without_duplicates
+
+
+def _validate_repository(repository: str) -> str:
+    """Validate a git repository."""
+    stripped = repository.strip()
+    parsed = urlparse(stripped)
+    if parsed.scheme != "http" and parsed.scheme != "https":
```

Comment:
suggestion(feel free to ignore):
```suggestion
    if parsed.scheme not in ["http", "https"]:
```